infra/ci/build.py

```diff
@@ -122,6 +122,26 @@ def should_build_coverage(project_yaml):
   return True
 
 
+def in_list(needle, haystack):
```

Comment:
A much shorter/possibly cleaner approach is to add a "flatten_options" function, which just iterates over the sanitizer list, and yields all the sanitizer names (handling dicts as appropriate). 

i.e. 

```python
def flatten_options(options):
  for item in options:
    if isinstance(item, dict):
      for key in item.keys():
        yield key
    else:
      yield item
```

Then below we just do

```python
os.getenv(env_var) in flatten_options(project_yaml.get(yaml_name, defaults))
```